### PR TITLE
Transfer ERC-20 only once per transaction

### PR DESCRIPTION
**Pull request by CodeSandwich**
Depends on https://github.com/radicle-dev/radicle-streaming/pull/50, closes https://github.com/radicle-dev/radicle-streaming/issues/37

### CODE REVIEW

File: src/EthPool.sol [hunk 1]
```diff
@@ -77,7 +77,11 @@ contract EthPool is Pool {
             );
     }
 
-    function _transfer(address to, uint128 amt) internal override {
-        if (amt != 0) payable(to).transfer(amt);
+    function _transfer(address userAddr, int128 amt) internal override {
+        // Take into account the amount already transferred into the pool
+        amt += int128(uint128(msg.value));
+        if (amt == 0) return;
+        require(amt > 0, "Sending a negative ether amount");
```
Comment (manuelpolzhofer) [inline src/EthPool.sol:84]:
```suggestion
        require(amt > 0, "not-enough-eth-sent");
```

Comment (manuelpolzhofer) [reply src/EthPool.sol:84]: Might be easier to understand for the user why it failed

Comment (manuelpolzhofer) [reply src/EthPool.sol:84]: eth transfer logic seems to work

**Review by manuelpolzhofer — APPROVED** [review]
looks good

File: src/EthPool.sol (hunk 1 above)
Comment (CodeSandwich) [reply src/EthPool.sol:84]: This is more of a sanity check and a message to the developer who breaks the transfer logic. It should never fail, the user can't break it, but can't fix it either, sending more eth won't help :D

**Pull request merged**